extendeddatasquare_test.go

```diff
@@ -59,119 +59,88 @@ func TestComputeExtendedDataSquare(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			result, err := ComputeExtendedDataSquare(tc.data, codec, NewDefaultTree)
+			result, err := ComputeExtendedDataSquare(tc.data, codec, DefaultTreeName)
 			assert.NoError(t, err)
 			assert.Equal(t, tc.want, result.squareRow)
 		})
 	}
 
 	t.Run("returns an error if chunkSize is not a multiple of 64", func(t *testing.T) {
 		chunk := bytes.Repeat([]byte{1}, 65)
-		_, err := ComputeExtendedDataSquare([][]byte{chunk}, NewLeoRSCodec(), NewDefaultTree)
+		_, err := ComputeExtendedDataSquare([][]byte{chunk}, NewLeoRSCodec(), DefaultTreeName)
 		assert.Error(t, err)
 	})
 }
 
 func TestImportExtendedDataSquare(t *testing.T) {
 	t.Run("is able to import an EDS", func(t *testing.T) {
 		eds := createExampleEds(t, shareSize)
-		got, err := ImportExtendedDataSquare(eds.Flattened(), NewLeoRSCodec(), NewDefaultTree)
+		got, err := ImportExtendedDataSquare(eds.Flattened(), NewLeoRSCodec(), DefaultTreeName)
 		assert.NoError(t, err)
 		assert.Equal(t, eds.Flattened(), got.Flattened())
 	})
 	t.Run("returns an error if chunkSize is not a multiple of 64", func(t *testing.T) {
 		chunk := bytes.Repeat([]byte{1}, 65)
-		_, err := ImportExtendedDataSquare([][]byte{chunk}, NewLeoRSCodec(), NewDefaultTree)
+		_, err := ImportExtendedDataSquare([][]byte{chunk}, NewLeoRSCodec(), DefaultTreeName)
 		assert.Error(t, err)
 	})
 }
 
 func TestMarshalJSON(t *testing.T) {
-	codec := NewLeoRSCodec()
-	result, err := ComputeExtendedDataSquare([][]byte{
-		ones, twos,
-		threes, fours,
-	}, codec, NewDefaultTree)
-	if err != nil {
-		panic(err)
-	}
+	original, err := ComputeExtendedDataSquare([][]byte{ones, twos, threes, fours}, NewLeoRSCodec(), DefaultTreeName)
+	require.NoError(t, err)
 
-	edsBytes, err := json.Marshal(result)
-	if err != nil {
-		t.Errorf("failed to marshal EDS: %v", err)
-	}
+	edsBytes, err := original.MarshalJSON()
+	require.NoError(t, err)
 
-	var eds ExtendedDataSquare
-	err = json.Unmarshal(edsBytes, &eds)
-	if err != nil {
-		t.Errorf("failed to marshal EDS: %v", err)
-	}
-	if !reflect.DeepEqual(result.squareRow, eds.squareRow) {
-		t.Errorf("eds not equal after json marshal/unmarshal")
-	}
+	var got ExtendedDataSquare
+	err = json.Unmarshal(edsBytes, &got)
+	require.NoError(t, err)
+
+	assert.Equal(t, original.dataSquare.Flattened(), got.dataSquare.Flattened())
+	assert.Equal(t, original.codec.Name(), got.codec.Name())
+	assert.Equal(t, original.treeName, got.treeName)
 }
 
-// TestUnmarshalJSON test the UnmarshalJSON function.
 func TestUnmarshalJSON(t *testing.T) {
-	treeName := "testing_unmarshalJSON_tree"
```

Comment:
[suggestion] It is more common to utilize slices of test structs, especially when different test cases share the same testing scenario, to prevent the repetition of the test scenario and to facilitate comparability of test cases. I'm curious about the reason behind this change in style.

Reply:
The previous test struct felt overly complicated because it included a step to malleate the input. On second thought, I could refactor this test back into a test case struct that avoids the malleation step. Will attempt that.

Reply:
95d949b5fe6c13d93d9e2f5e468675057ab7a854